sound/soc/intel/boards/Kconfig

```diff
@@ -448,7 +449,7 @@ if SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC || SND_SOC_SOF_HDA_AUDIO_CODEC
 
 config SND_SOC_INTEL_SKL_HDA_DSP_GENERIC_MACH
 	tristate "SKL/KBL/BXT/APL with HDA Codecs"
-	depends on SND_HDA_CODEC_HDMI
+	depends on SND_HDA_CODEC_HDMI && SND_SOC_SOF_HDA_AUDIO_CODEC
```

Comment:
This is not needed as SND_SOC_SOF_HDA_AUDIO_CODEC is already in the "if" statement above.

Reply:
@kv2019i In the "if" statement, it requires "if SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC || SND_SOC_SOF_HDA_AUDIO_CODEC"
This means SND_SOC_SOF_HDA_AUDIO_CODEC is not a "must". Let's look into the first condition "SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC". In its description, "This option is DEPRECATED". And I checked the src code, This config is only used in SST driver now. So I believe the "if" statement is expired. We should just use "if SND_SOC_SOF_HDA_AUDIO_CODEC".

Reply:
Discussion ongoing on that deprecation, we can't remove it now. I think the if check is still right. If SST is used (SKYLAKE_HD_AUDIO_CODEC part of the if check), the machine driver will not use common HDMI driver but will use hdac-hdmi. If SOF is used (the other part of the if), the machine driver will always set up common codec and SND_HDA_CODEC_HDMI is needed.

For simplicity, SND_HDA_CODEC_HDMI is selected always, but to be really accurate, it should only be a dependency, if SOF is enabled in the build.

I don't think you need to modify this.

Reply:
@kv2019i If so, I think we have 2 choice:
1. use "if SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC || (SND_SOC_SOF_HDA_AUDIO_CODEC && SND_HDA_CODEC_HDMI)" in the if statement, or
2. we don't change the if statement and use 'depends on SND_HDA_CODEC_HDMI && SND_SOC_SOF_HDA_AUDIO_CODEC' which is used in my original patch.
Or do you have other suggestions?

For the first choice, I still have a tiny concern. It is possible that SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC is enabled and (SND_SOC_SOF_HDA_AUDIO_CODEC && SND_HDA_CODEC_HDMI) are not both enabled. 
So in this case, SND_SOC_SOF_HDA_AUDIO_CODEC can be disabled and SND_HDA_CODEC_HDMI is enabled, and SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC is enabled while we are still using SOF driver. This will be wrong.

I'm not sure if this option is acceptable: let's remove the 'if' statement and use:
depends on SND_SOC_SOF_HDA_AUDIO_CODEC
depends on SND_HDA_CODEC_HDMI

Reply:
@libinyang I'd vote for option

3. don't change the definitions at all, I think this is ok for SST and SOF

I think only problem is that SND_HDA_CODEC_HDMI needs to be selected to compile for SST, but all distributions have SND_HDA_CODEC_HDMI enabled (due to legacy), so this seems a pretty minor problem. Code will still work, worst that can happen is that you can't select SST.

For the other options, I think the drawbacks are bigger:
1. I think you are right that iIf SOC_INTEL_SKYLAKE_HDAUDIO_CODEC and SOF_HDA_AUDIO_CODEC are defined but not HDA_CODEC_HDMI, and you use this on a system with dmics, you will hit an error at runtime as HDMI probe will fail. This seems much worse than the drawback of option 3.

2. This means that to build SST SKL driver, you'd need to also enable SOF in kernel config. This seems a bigger problem than (3) as SST and SOF are not typically enabled in distro configs by default and it is very likely if you are enabling SST driver, you will miss some SOF options.

I still think (3) is least bad.